insights/tests/client/data_collector/test__process_content_redaction.py

```diff
@@ -46,47 +44,18 @@ def test_subproc_calls_fgrep(tmpfile, Popen):
     tmpfile.return_value.write.assert_called_once_with('\n'.join(['test1', 'test2']).encode('utf-8'))
     tmpfile.return_value.flush.assert_called_once()
     Popen.assert_has_calls([
-        call(['sed', '-rf', constants.default_sed_file, test_file.name], stdout=PIPE),
-        call(['grep', '-v', '-F', '-f', tmpfile.return_value.name], stdin=Popen.return_value.stdout, stdout=PIPE)
+        call(['grep', '-v', '-F', '-f', tmpfile.return_value.name, test_file.name], stdout=PIPE)
     ])
 
 
-@patch('insights.client.data_collector.Popen')
-@patch('insights.client.data_collector.NamedTemporaryFile')
-def test_nogrep(tmpfile, Popen):
-    '''
-    Verify that grep is not called when no patterns to exclude
-    are present
-    '''
-    Popen.return_value.communicate = Mock(return_value=('test', None))
-    _process_content_redaction(test_file.name, None, False)
-    tmpfile.assert_not_called()
-    Popen.assert_called_once_with(['sed', '-rf', constants.default_sed_file, test_file.name], stdout=PIPE)
-
-
-# mock the .exp.sed file for QE pipeline
-mock_sed_file = NamedTemporaryFile()
-mock_sed_file.write("s/(password[a-zA-Z0-9_]*)(\\s*\\:\\s*\\\"*\\s*|\\s*\\\"*\\s*=\\s*\\\"\\s*|\\s*=+\\s*|\\s*--md5+\\s*|\\s*)([a-zA-Z0-9_!@#$%^&*()+=/-]*)/\\1\\2********/\ns/(password[a-zA-Z0-9_]*)(\\s*\\*+\\s+)(.+)/\\1\\2********/".encode('utf-8'))
-mock_sed_file.flush()
-
-
-@patch('insights.client.data_collector.constants.default_sed_file', mock_sed_file.name)
 def test_returnvalue():
     '''
     Verify that the returned data is what we expect to see
     '''
-    # no exclude
-    retval = _process_content_redaction(test_file.name, [], False)
-    assert retval == 'test\nabcd\n1234\npassword: ********\n'.encode('utf-8')
-
-    # no exclude also works with None
-    retval = _process_content_redaction(test_file.name, None, False)
-    assert retval == 'test\nabcd\n1234\npassword: ********\n'.encode('utf-8')
-
     # exclude plainstrings
     retval = _process_content_redaction(test_file.name, ['test', 'abc'], False)
-    assert retval == '1234\npassword: ********\n'.encode('utf-8')
+    assert retval == '1234\npassword: pAsswOrd\n'.encode('utf-8')
 
     # exclude regex
     retval = _process_content_redaction(test_file.name, ['[[:digit:]]+', 'a*(b|c)'], True)
-    assert retval == 'test\npassword: ********\n'.encode('utf-8')
+    assert retval == 'test\npassword: pAsswOrd\n'.encode('utf-8')
```

Comment:
@ryan-blakley, yes, we want to do that, since the `password` redaction caused some issues (#3858, #3885, and another not-reported one in the `insalled_rpms`) and resulted in false positives of some rules. 
and, the current logic of the `.exp.sed` is not sufficient, e.g. the keyword `password` is not the only one for indicating passwords. Considering these, and we have a strict review process for Specs, We'd like to remove it from the collection. What do you think?

Reply:
Ps, the `_process_content_redaction` method will be removed also in the next `Move obfuscation/reaction to core collection` (#3679). The relevant tests will also be removed then.

Reply:
Gotcha yeah I noticed the issues that were caused by it. Is the plan to not merge this until the new obfuscation/redaction is merged? The reason I ask, I just don't want passwords to get stored in warehouse after making this change, and then have to purge them.

Reply:
Sorry @ryan-blakley, this `.exp.sed` or the equivalent process will not be used/implemented in the new obfuscation/redaction. Our current target is to completely discard it from the core (and count on spec reviews to avoid collecting PII information) due to 1. it causes issues on specs 2. it's not sufficient. 

However, if you have come across any real cases where this "password" script works as expected, for example, actually obfuscated the password, can you share some of these cases with me? 

We have talked about the above in the sync meeting too. I believe Bob will also ask you the same for me. It won't hurt that we start another e-mail or Slack thread to continue discussing this matter.

Reply:
@xiangce Gotcha, I apologize I thought the current `.exp.sed` script was actively redacting passwords. Looking through several older archives I have downloaded I see how bad it is now. I did see it redacted the proxy password from the /etc/rhsm/rhsm.conf file correctly, but that's not as big of a deal. My fear was we would start collecting service user passwords, or database user passwords for like mariadb, redis, etc. But if that's not the case, then the change looks good to me!

Reply:
@ryan-blakley - Thanks for the info. So I think it's required for us to go through the existing (`non-filterable`) specs to check if there are any risks of collecting the sensitive information. We can merge this PR after this check.

And I opened #3916 first for `rhsm.conf`

Reply:
@ryan-blakley - FYI, I opened INSGHTCORE-236 to track the password check of existing specs.

Reply:
Sounds perfect to me!